[23기 한상웅] Refactor:리뷰반영(컨벤션)

tkddnd82 opened the pull request:
-좋아요 기능 추가(로그인시)

>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 로그인 버튼 활성화 기능 구현(예시)
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- id '@' 포함, pw 5 글자 이상 입력 시 버튼 활성화(예시)
![wishlist1](https://user-images.githubusercontent.com/84978912/128808136-e10fa130-89c3-4d72-874f-9494cedd9d5a.jpg)
![wishlist2](https://user-images.githubusercontent.com/84978912/128808146-c3ab9ab6-cc9e-446a-a651-bb5cabbb9aab.jpg)
![wishlist3](https://user-images.githubusercontent.com/84978912/128808152-d73c98b1-a6d8-41e8-8aa5-3b9d836386b5.jpg)
![wishlist4](https://user-images.githubusercontent.com/84978912/128894606-3b561806-adba-4998-b70c-a216262b1ac0.jpg)


## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 함수 로직이 너무 긴 것 같습니다. 좀 더 효율적인 방법이 없을지 궁금합니다.(예시) 

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x ] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x ] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [ x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

tkddnd82 committed:
Add:좋아요(wishlist)기능 추가
-좋아요 기능 추가(로그인시)

S2-J1NG commented on wishlists/views.py:
```diff
@@ -1,3 +1,47 @@
+                request.user.product_set.remove(product_id)
+
+            else:    
+                request.user.product_set.add(product_id)
```
`add` 말고 `create` 으로 해서 적용해보세요!

ghost commented:
통신결과 PR 에 꼭 첨부해주세요!

tkddnd82 committed:
qMerge branch 'main' into feature/wishlist

tkddnd82 committed:
Refactor:
-remove대신 delete사용/add대신 create로 수정
-Q객체 filter에 적용
-띄어쓰기 관련 수정
-통신결과 PR에 첨부

tkddnd82 committed:
Merge branch 'main' into feature/wishlist

S2-J1NG commented on wishlists/views.py:
```diff
@@ -1,3 +1,52 @@
+            user=User.objects.get(id=request.user.id)
+            like_list      = user.wishlist_set.all()
+            
+            like_items = [{
+            'product_id'  : product.id,
+            'name'        : product.name, 
+
+        }for product in like_list]
```
`product_id` 와 `product_name` 만 프론트로 전달하면 되나요?? 제품 이미지나 가격은 전달 안 해도 될까요??

tkddnd82 replied:
멘토님 조언으로 product_name, img, price 추가적용하였습니다. 
price 추가하면서 annotate 적용하였습니다. 개인적으로 이해가 오래걸려서 시간이 걸렸습니다.
위시리스트 페이지에서 가격은 미니멈 가격이 적용되어서 가격은 미니멈 가격으로 설정되도록 하였습니다.

S2-J1NG replied:
상웅님~! 추가한 내용 코드가 없는 거 같은데 혹시 `push` 하셨을까요?

tkddnd82 replied:
어제 저녁에 계속 오류가 생겨서 방금 push했습니다 :)

ghost commented:
통신 결과 첨부 👍 리뷰 내용 프론트와 확인해서 반영해주세요!

tkddnd82 committed:
Refactor:수정사항 적용
-like_item리스트에서 name,image,price추가
-price추가하면서 annotate 적용 위시리스트 페이지내에서 가격은 미니멈 가격으로
나와서 미니멈으로 나올수 있다록 MIn import함

tkddnd82 committed:
Refactor:리뷰사항 적용
-like_items리스트에서 product_name,img,price추가
-위시리스트페이지에서 가겨은 미니멈으로 설정되어서 min import함
-price추가하면서 annotate()적용

tkddnd82 committed:
Refactor:리뷰사항 적용
-like_items리스트에서 product_name,img,price추가
-위시리스트페이지에서 가겨은 미니멈으로 설정되어서 min import함
-price추가하면서 annotate()적용

S2-J1NG commented on wishlists/views.py:
```diff
@@ -1,3 +1,53 @@
-from django.shortcuts import render
+import json
+from my_settings import SECRET_KEY
 
-# Create your views here.
+from wishlists.models import Wishlist
+
+from django.http            import JsonResponse
+from django.views           import View
+from django.db.models       import Q, Min
+from products.models        import (Category,SubCategory,Product,Option,ProductDetailImage,Tag,)
+from users.utils             import login_required 
+from users.models            import User
```
`import` 와 `from` 사이 줄 띄워주세요!
위코드 컨벤션에 나와 있으니 찾아서 적용해주세요! 같은 프로젝트에서는 동일한 컨벤션으로 코드 정렬해야 합니다 :)

tkddnd82 replied:
넵!

S2-J1NG commented on wishlists/views.py:
```diff
@@ -1,3 +1,53 @@
+            user=User.objects.get(id=request.user.id)
+            like_list      = user.wishlist_set.all().annotate(price=Min("product__option__price"))
+            
+            like_items = [{
+            'product_id'   : like.product.id,
+            'name'         : like.product.name,
+            'image'        : like.product.thumbnail_image_url,
+            'price'        : like.price
+            }for like in like_list]
```
```
like_items = [{
      'product_id' : like.product.id,
      'name'           : like.product.name,
      'image'          : like.product.thumbnail_image_url,
      'price'          : like.price} for like in like_list] 
```

S2-J1NG replied:
코드 컨벤션 지켜주세요!

S2-J1NG commented on wishlists/views.py:
```diff
@@ -1,3 +1,53 @@
+            }for like in like_list]
+        except KeyError:
+            return JsonResponse({'Message' : 'KEY_ERROR'}, status = 400)
+        return JsonResponse({'Like_Items' : like_items}, status = 200)
```
`try` 에 대한 `return`은 `except` 위에 위치 시켜주세요!

S2-J1NG commented on wishlists/views.py:
```diff
@@ -1,3 +1,53 @@
+            if Wishlist.objects.filter(Q(product_id=product_id) & Q(user_id=user_id)).exists():
+               Wishlist.objects.filter(product_id=product_id).delete()
+               return JsonResponse({'Message' : 'SUCCESS_DELETE'}, status = 200)
+        
+
```
불필요한 공백은 지워주세요! 공백도 모두 연산 처리하게 때문에 불필요한 메모리가 사용됩니다~!

S2-J1NG commented on wishlists/views.py:
```diff
@@ -1,3 +1,53 @@
+        
+
+            else:
+                Wishlist.objects.create(
+                product_id    = data['product_id'],
+                user_id       = user_id
+
+                ) 
```
여기도 불필요한 줄 삭제해주세요

S2-J1NG commented on wishlists/views.py:
```diff
@@ -1,3 +1,53 @@
+        except KeyError:
+            return JsonResponse({'Message' : 'KEY_ERROR'}, status = 400)
+
+        return JsonResponse({'Message' : 'SUCCESS_CREATE'}, status = 200)
```
`try` 에 대한 `return` 위치 변경해주세요~!

ghost commented:
기능 구현 👍  전체적인 코드 컨벤션 확인해서 반영해주세요!

tkddnd82 commented:
> 기능 구현  전체적인 코드 컨벤션 확인해서 반영해주세요!

넵 컨벤션 주의하도록 하겠습니다!

tkddnd82 committed:
Refactor: 리뷰 반영
-전체적으로 컨벤션 맞춰서 수정

ghost commented:
LGTM ✨
